src/components/sign-transactions-modal/components/sign-transaction-component/sign-transaction-component.tsx

```diff
@@ -6,14 +6,87 @@ import state from '../../signTransactionsModalStore';
 @Component({
   tag: 'sign-transaction-component',
   styleUrl: 'sign-transaction-component.css',
-  shadow: false,
+  shadow: false
 })
 export class SignTransaction {
   @Prop() header: VNode;
 
+  getSignButtonProps() {
+    const { currentIndex, nextUnsignedTxIndex } = state.commonData;
+
+    if (currentIndex === nextUnsignedTxIndex) {
+      return {
+        signText: 'Sign',
+        disabled: state.isWaitingForSignature,
+        'data-testid': DataTestIdsEnum.signTransactionBtn,
+        onClick: state.onSign
+      };
+    }
+
+    return {
+      signText: 'Next',
+      disabled: false,
+      'data-testid': DataTestIdsEnum.signNextTransactionBtn,
+      onClick: state.onNext
+    };
+  }
+
+  getBackButtonProps() {
+    const { transactionsCount, currentIndex } = state.commonData;
+    const isMultipleTransactions = transactionsCount > 1;
+
+    if (!isMultipleTransactions) {
+      return null;
+    }
+
+    if (currentIndex === 0) {
+      return {
+        'data-testid': DataTestIdsEnum.signCancelBtn,
+        backButtonText: 'Cancel',
+        onClick: state.onCancel
+      };
+    }
+
+    return {
+      'data-testid': DataTestIdsEnum.signBackBtn,
+      backButtonText: 'Back',
+      onClick: state.onPrev,
+      disabled: state.isWaitingForSignature
+    };
+  }
+
+  getHighlightedData() {
+    const { data, highlight } = state.commonData;
+
+    if (!highlight || !data) {
+      return data;
+    }
+
+    const parts = data.split(highlight);
+
+    return (
+      <span>
+        {parts.map((part, index) => (
+          <span>
+            <span class="data-content">{part}</span>
+            {index < parts.length - 1 && (
+              <span class="highlight">{highlight}</span>
+            )}
+          </span>
+        ))}
+      </span>
+    );
+  }
+
   render() {
-    const { receiver, egldLabel, data, feeInFiatLimit, feeLimit } = state.commonData;
+    const { receiver, egldLabel, feeInFiatLimit, feeLimit, scCall } =
+      state.commonData;
 
+    const { signText, ...signButtonProps } = this.getSignButtonProps();
+    const { backButtonText, ...backButtonProps } =
+      this.getBackButtonProps() ?? {};
```

Comment:
why not return empty object from function without having to provide here a fallback ?